test/c/deadlock0.c

```diff
@@ -76,11 +77,14 @@ void taskC(void *args) {
 }
 
 void taskMain(void *args) {
-    promise_list = hclib_promise_create_n(1, true);
+    promise_list = (hclib_promise_t **)malloc(2 * sizeof(hclib_promise_t *));
     future_list = (hclib_future_t **)malloc(1 * sizeof(hclib_future_t *));
-    promise = hclib_promise_create();
-    promise_list[0] = promise;
+    assert(promise_list && future_list);
+
+    promise_list[0] = hclib_promise_create();
+    promise_list[1] = NULL;
     future_list[0] = hclib_get_future_for_promise(promise);
```

Comment:
This looks like a bug. You're trying to extract a future from the promise variable, which is uninitialized at this point